sparta/src/Report.cpp

```diff
@@ -1159,6 +1159,157 @@ class ReportFileParserYAML
     std::string filename_; //!< For recalling errors
 }; // class ReportFileParserYAML
 
+Report::StatAdder Report::add(const StatisticInstance& si, const std::string& name) {
+    if(name != "" && stat_names_.find(name) != stat_names_.end()){
+        throw SpartaException("There is already a statistic instance in this Report (")
+            << getName() << ") named \"" << name << "\" pointing to "
+            << getStatistic(name).getLocation()
+            << " and the new stat would be pointing to a StatisticInstance "
+            << si.getExpressionString();
+    }
+
+    // Track a new stat with helpful exception wrapping
+    addField_(name, si);
+
+    if(name != ""){ stat_names_.insert(name); }
+
+    addSubStatistics_(&si);
+
+    return Report::StatAdder(*this);
+}
```

Comment:
`Report::add` might be worth templatizing. It looks like several of the overrides share the same code.

Reply:
I 100% agree with you.  However, I think I'll save that for another day.  This is code I originally didn't write/own, and I just did some pretty big surgery on it.

Reply:
New Issue opened